external/webserver/http_client.c

```diff
@@ -1150,6 +1150,7 @@ int http_send_response_helper(struct http_client_t *client, int status, const ch
                                                            body_len);
 				if (len < 0) {
 					HTTP_LOGE("Error: snprintf failed \n");
+					HTTP_FREE(buf);
```

Comment:
how about adding goto for the error cases?
```
goto error;
```
```
error :
	HTTP_FREE(buf);
	return HTTP_ERROR;
```

Reply:
I prefer to avoid using "goto" statement.